### PR TITLE
Jeremy/eng 957 mobile listing form

### PR DESCRIPTION
**Pull request by jerminatorhits**
## Description
Why did you write this code?
https://beetoken.atlassian.net/browse/ENG-957

## How to Test
- [ ] Visit https://dev02.beenest.io/host/listings/87134/listing_info
- [ ] Try filling out the form and navigating tabs on all screen sizes, esp mobile

Which devices did you test on?
- [x] Chrome on Mac
- [ ] Chrome on PC
- [ ] Firefox on Mac
- [ ] Firefox on PC
- [x] Safari iPhone
- [ ] Chrome Android

## REVIEWERS:
Check against these principles:

### High level
Does this code need to be written?
What are the alternatives?
Will this implementation become a support issue?
How much error margin does this solution have?

### Code
* Does the code follow industry standards?
JS: https://github.com/airbnb/javascript
React: https://github.com/airbnb/javascript/tree/master/react
https://github.com/vasanthk/react-bits
Documentation headers: http://usejsdoc.org/index.html

* Is there duplicated code? Can it be refactored into a shared method?
* Is the code consistent with our project?
* Are there unit tests? Do they test the states?
* Is the person refactoring another developer's code? If possible, did the original developer approve?

### Variables/Naming:
* Would the variable type led to future edge cases?
* Are the variable naming clear? Would the value contain something other than what the name describes.

### Security
* Can this be hacked or abused by the user?

## Further Work
Will this have future work?

## Learnings
Did you learn anything here you want to share with the team?
hyphens: auto is a thing. auto breaks words using a dictionary of sorts.
https://css-tricks.com/almanac/properties/h/hyphenate/

### CODE REVIEW

File: src/components/routes/Host/ListingForm/ListingForm.container.ts [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
+const ListingFormMobileContainer = styled.section`
   width: 100%;
 
-
   & > .bee-general-wrapper {
-    min-height: calc(100% - 64px);
     margin: 0 auto;
+    margin-bottom: 56px;
+    min-height: calc(100% - 64px);
+    min-width: auto;
```
Comment (kcvan) [inline src/components/routes/Host/ListingForm/ListingForm.container.ts]: i dont think `auto` is a valid value for min-width, i think you're looking for `0`, but i also think you don;'t need to add it if it's not being set anywhere else?

File: src/components/routes/Host/ListingForm/ListingForm.container.ts [hunk 2]
```diff
@@ -96,4 +95,26 @@ const ListingFormContainer = styled.section`
...
+    }
+  }
+`;
+
+const ListingFormDesktopContainer = styled(ListingFormTabletContainer)`
+  @media (min-width: 1025px) {
+    & > .bee-general-wrapper {
+      width: 976px;
+      form {
+        padding-left: 0;
```
Comment (kcvan) [inline src/components/routes/Host/ListingForm/ListingForm.container.ts]: for conformity, i would use `padding: 40px 56px 96px 0;`

File: src/components/routes/Host/ListingForm/ListingForm.tsx [hunk 3]
```diff
@@ -192,7 +192,6 @@ class ListingForm extends React.Component<Props, State> {
                     onClick={() => {
                       if (!FormikProps.isValid) {
                         alert(`Cannot save changes due to errors:\n\n${Object.values(FormikProps.errors).join('\n').toString()}`);
```
Comment (kcvan) [inline src/components/routes/Host/ListingForm/ListingForm.tsx]: might need to change`\n` to `\n\r` fo windows

File: src/components/routes/Host/ListingForm/ListingFormNav/ListingFormNav.container.ts [hunk 4]
```diff
@@ -1,72 +1,38 @@
...
+  width: 100%;
 
   .bee-general-wrapper {
     display: flex;
+    flex-direction: column;
     justify-content: space-between;
     height: 100%;
     margin: 0 auto;
-
+    min-width: auto;
```
Comment (kcvan) [inline src/components/routes/Host/ListingForm/ListingFormNav/ListingFormNav.container.ts]: same here

File: src/components/routes/Host/ListingForm/ListingForm.container.ts (hunk 2 above)
Comment (tc) [reply src/components/routes/Host/ListingForm/ListingForm.container.ts]: if this is a consistent padding for all form containers, can it be a variable somewhere or just applied to `form` generally?

File: src/components/routes/Host/ListingForm/ListingForm.tsx (hunk 3 above)
Comment (jerminatorhits) [reply src/components/routes/Host/ListingForm/ListingForm.tsx]: so with just using `\n` it looks like:

image

but with `\n\r` it looks like there are double spaces:

image

File: src/components/routes/Host/ListingForm/ListingFormNav/ListingFormNav.container.ts (hunk 4 above)
Comment (jerminatorhits) [reply src/components/routes/Host/ListingForm/ListingFormNav/ListingFormNav.container.ts]: removed width="100%" as props passed to GeneralWrapper and declared `width` and `min-width` here as well

**Comment by kcvan** [comment]
running into stuff looking like this:
screen shot 2019-02-04 at 12 30 29 pm

1) the tabs look a little tall, maybe `64px`?
2) i think `Checkin` should be `Check-in`
3) we should allocate a space for the `Save & Exit` link so it's not on the same line as the first input titles
4) the form can use `padding: 48px 24px 56px;` (this might fix #3)
5) we can remove the `margin-bottom: 56px;` on the bee general wrapper because the new padding bottom from #4 will add this

File: src/components/routes/Host/ListingForm/ListingForm.container.ts (hunk 2 above)
Comment (jerminatorhits) [reply src/components/routes/Host/ListingForm/ListingForm.container.ts]: I think the `padding` was for the layout of this particular form -- though that would be something good to keep in mind a `<BeeFormContainer>` that has all the layouts/styles for a basic form.

**Comment by jerminatorhits** [comment]
@kcvan could not replicate `1.` but made adjustments to accommodate the extra margin from `6.`
All other items should be updated

**Pull request merged**